lib/infrastructure-stack.ts

```diff
@@ -1,16 +1,127 @@
+        blockPublicAcls: true,
+        ignorePublicAcls: true,
+        blockPublicPolicy: false,
+        restrictPublicBuckets: false,
+      }),
+      removalPolicy: cdk.RemovalPolicy.RETAIN,
+    });
 
-    // example resource
-    // const queue = new sqs.Queue(this, 'InfrastructureQueue', {
-    //   visibilityTimeout: cdk.Duration.seconds(300)
-    // });
+    new s3.Bucket(this, "Dnd5eApiCoBucket", {
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      websiteRedirect: {
+        hostName: "www.dnd5eapi.co",
+        protocol: s3.RedirectProtocol.HTTPS,
+      },
+      removalPolicy: cdk.RemovalPolicy.RETAIN,
+    });
+
+    // Reference the active ACM certificate in us-east-1 for CloudFront
+    const certificate = acm.Certificate.fromCertificateArn(
+      this,
+      "ActiveAcmCertificateForCloudFront",
+      "arn:aws:acm:us-east-1:911448592982:certificate/b08418e0-443b-408d-9094-ba6e716ede2b"
+    );
+
+    // Reference the existing Route 53 Hosted Zone
+    const hostedZone = route53.HostedZone.fromHostedZoneAttributes(
+      this,
+      "ImportedHostedZone",
+      {
+        hostedZoneId: "ZDMYNHE4G4KLW",
+        zoneName: "dnd5eapi.co",
+      }
+    );
+
+    // Define the Response Headers Policy to match the existing configuration
+    const customResponseHeadersPolicy = new cloudfront.ResponseHeadersPolicy(
+      this,
+      "DndResponseHeadersPolicy",
+      {
+        responseHeadersPolicyName: "DndAppResponseHeadersPolicy",
+        comment:
+          "Allows all origins for CORS requests, including preflight requests - matches existing policy",
+        corsBehavior: {
+          accessControlAllowOrigins: ["*"],
+          accessControlAllowHeaders: [],
+          accessControlAllowMethods: [
+            "GET",
+            "HEAD",
+            "PUT",
+            "POST",
+            "PATCH",
+            "DELETE",
+            "OPTIONS",
+          ],
+          accessControlAllowCredentials: false,
+          accessControlExposeHeaders: ["*"],
+          originOverride: false,
+        },
+      }
+    );
+
+    // Define s3WebsiteOrigin before it's used in policies or distribution
+    const s3WebsiteOrigin = new origins.HttpOrigin(
+      "dnd5eapi.co.s3-website-us-west-1.amazonaws.com",
+      {
+        protocolPolicy: cloudfront.OriginProtocolPolicy.HTTP_ONLY,
+      }
+    );
+
+    // Cache Policy - to handle QueryString: true and QueryStringCacheKeys: ["nope"]
+    const cachePolicy = new cloudfront.CachePolicy(this, "DndCachePolicy", {
+      cachePolicyName: "DndAppCachePolicy",
+      comment:
+        "Cache policy for dnd5eapi.co to forward all query strings but only cache on nope",
+      cookieBehavior: cloudfront.CacheCookieBehavior.none(),
+      headerBehavior: cloudfront.CacheHeaderBehavior.none(),
+      queryStringBehavior:
+        cloudfront.CacheQueryStringBehavior.allowList("nope"),
+      minTtl: cdk.Duration.seconds(0),
+      defaultTtl: cdk.Duration.seconds(86400),
```

Comment:
Nit: make a constant for these seconds for what this translates to

Reply:
Apparently, you can just do `cdk.Duration.days(1)` and `cdk.Duration.days(365)`